File: April/Abiji-2020/C++/Time Needed to Buy Tickets/Time Needed to Buy tickets.cpp
class Solution {
public:
    int timeRequiredToBuy(vector<int>& tickets, int k) {
        int ans=0;
        int i=0;
        while(tickets[k]!=0){
            if (tickets[i]!=0){
                ans++;
                tickets[i]--;
            }
            i++;
            if(i==tickets.size()) i=0;
        }
        return ans;
    }
};

File: April/Abiji-2020/Golang /Time Needed to Buy Tickets/Time Needed to Buy tickets.go
func timeRequiredToBuy(tickets []int, k int) int {
    ans :=0
    i:=0
    for tickets[k]!=0{
        if tickets[i]!=0{
            ans++
            tickets[i]--
        }
        i++
        if i==len(tickets) {
            i=0
        }
    }
    return ans
}
